### PR TITLE
Allow custom fonts and font sizes

### PR DESCRIPTION
**Pull request by robertdown**
This allows users to select their font-family and font-size in the preferences. Couple of things left to do. First, we use a LOT of inline CSS that have to come out for this to be completely effective. Ironically, the globals page suffers from this and it won't change (Well, the tabs will, content won't).

I know we've had CSS values in the globals before and I understand _why_ we took them out, but these two values (at minimum) should definitely be a per-user option (I think @sjpadgett mentioned it on a separate thread recently).

I don't have everything worked out yet (Though you can totally test this and it'll work). For instance, right now there's `themeBuilder.php` and `themeBuilder.css`. The CSS acts as a template for the php file which loops over possible custom options and does a `str_replace`, and then inside of (eventually core.css) style_light.css it `@imports url(themeBuilder.php);` which sets the content-type to text/css. There's probably a more elegant solution, but wanted to get this on the radar.

You'll see a font called "Lato" which is an open source font that I'm going to embed so we've got access to it (It's a great font and will be the Light themes new default here soon.

### CODE REVIEW

File: interface/themes/core.css [hunk 1]
```diff
@@ -8,6 +8,8 @@
  * @license https://github.com/openemr/openemr/blob/master/LICENSE General Public License 3
  */
 
+@import "themeBuilder.php";
```
Comment (bradymiller) [inline interface/themes/core.css]: this line kind of blows my mind :)

File: interface/themes/style_light.css [hunk 2]
```diff
@@ -1506,22 +1446,21 @@ table.dataTable#therapy_groups_list tbody td{
...
 
 .sidebar ul.mainmenu {
     padding-bottom: 55px;
 }
 
 .alert-danger {
-  font-size: 100% !important;
   background-color: #FF0000 !important;
   color: #fff !important;
 }
```
Comment (bradymiller) [inline interface/themes/style_light.css]: will this sort of thing need to be done to all the themes?

File: library/globals.inc.php [hunk 3]
```diff
@@ -172,6 +172,37 @@ function gblTimeZones()
...
+            xl('Default font size'),
+            [
+                '__default__' => 'Use Theme Font Size',
+                '10' => '10pt',
+                '12' => '12pt',
+                '14' => '14pt',
+                '16' => '16pt',
+                '18' => '18pt',
+            ],
+            'Use THeme Font Size',
```
Comment (bradymiller) [inline library/globals.inc.php]: change THeme to theme

Comment (bradymiller) [inline library/globals.inc.php]: actually, this should be '__default__'

File: library/globals.inc.php [hunk 4]
```diff
@@ -172,6 +172,37 @@ function gblTimeZones()
...
+                'Arial, Helvetica, sans-serif' => "Arial",
+                '"Arial Black", Gadget, sans-serif' => "Arial Black",
+                'Impact, Charcoal, sans-serif' => "Impact",
+                '"Lucida Sans Unicode", "Lucida Grande", sans-serif' => "Lucida Sans",
+                'Tahoma, Geneva, sans-serif' => "Tahoma",
+                '"Trebuchet MS", Helvetica, sans-serif' => "Trebuchet MS",
+                'Verdana, Geneva, sans-serif' => "Verdana",
+                'Lato' => "Lato",
+            ],
+            'Use theme font',
```
Comment (bradymiller) [inline library/globals.inc.php]: this should be '__default__'

File: interface/themes/style_light.css (hunk 2 above)
Comment (robertdown) [inline interface/themes/style_light.css]: Well, percentages can stay. Any hard typed values (Such as 14pt) will have to be changed. The goal is to get all inline styling out and into CSS

**Comment by bradymiller** [comment]
Am curious to hear @sjpadgett 's thoughts on this.

If this is working after above issues are addressed(and no showstoppers per @sjpadgett ), then seems like ok to bring this into the codebase (ie. can further develop it and work it into the other themes down the road).

File: library/globals.inc.php [hunk 5]
```diff
@@ -172,6 +172,37 @@ function gblTimeZones()
             xl('Pick a general theme (need to logout/login after change this setting).')
         ),
 
+        'font-family' => [
+            xl('Default font'),
+            [
+                '__default__' => 'Use Theme Font',
+                'Arial, Helvetica, sans-serif' => "Arial",
+                '"Arial Black", Gadget, sans-serif' => "Arial Black",
```
Comment (bradymiller) [inline library/globals.inc.php]: bad quote " here

Comment (bradymiller) [inline library/globals.inc.php]: hmm actually, not sure what this is. For some reason, you have " around some terms and not others?
I am guessing this is needed for the css. If so, then disregard.

Comment (robertdown) [inline library/globals.inc.php]: It's supposed to be there

**Comment by bradymiller** [comment]
If you comfortable with this as a starting point and it's not breaking anything, then this seems reasonable to bring into the codebase.

**Comment by robertdown** [comment]
Something's not right with it, I'll look over the weekend. Probably something minor

**Pull request merged**